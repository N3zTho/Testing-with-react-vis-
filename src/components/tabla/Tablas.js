import React from "react";

import TablaYear from "./TablaYear";

const Tablas = ({ years, range }) => {
  return (
    <>
      <div className="table-responsive">
        {years.map(tabla => (
          <table className="table table-borderless table-sm" key={tabla.year}>
            <tbody>
              <tr>
                <td>
                  <table className="table table-bordered">
                    <thead className="thead-light">
                      <tr>
                        <th>&nbsp;</th>
                      </tr>
                      <tr>
                        <th>Denominacion</th>
                      </tr>
                      <tr>
                        <td>Serie desestacionalizada</td>
                      </tr>
                      <tr>
                        <td>Tendencia ciclo</td>
                      </tr>
                    </thead>
                  </table>
                </td>
                <td>
                  <TablaYear tabla={tabla} range={range} />
                </td>
              </tr>
            </tbody>
          </table>
        ))}
      </div>
    </>
  );
};

export default Tablas;
