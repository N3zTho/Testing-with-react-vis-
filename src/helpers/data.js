const data = {
  2006: {
    serie: "3.2|3.5|3.3|3.4|3.1|3.5|3.7|3.6|3.7|3.7|3.7|3.8".split("|"),
    tendencia: "3.2|3.2|3.3|3.4|3.5|3.5|3.6|3.6|3.7|3.7|3.7|3.8".split("|")
  },

  2007: {
    serie: "3.7|3.8|3.8|3.7|3.5|3.5|3.5|3.6|3.5|3.6|3.6|3.5".split("|"),
    tendencia: "3.8|3.8|3.7|3.7|3.6|3.6|3.5|3.5|3.5|3.6|3.6|3.6".split("|")
  },
  2008: {
    serie: "3.8|3.7|3.8|3.6|3.6|3.6|3.8|3.8|3.9|4|4.5|4.5".split("|"),
    tendencia: "3.6|3.7|3.7|3.7|3.7|3.7|3.7|3.8|3.9|4.1|4.3|4.5".split("|")
  },
  2009: {
    serie: "4.7|5|5|5.2|5.6|5.3|5.5|5.7|6|5.5|5.3|5.3".split("|"),
    tendencia: "4.7|4.9|5.1|5.3|5.5|5.7|5.6|5.6|5.5|5.5|5.4|5.3".split("|")
  },
  2010: {
    serie: "5.4|5.1|5.1|5.5|5.3|5.1|5.3|5|5.3|5.3|5.3|5.5".split("|"),
    tendencia: "5.3|5.3|5.3|5.2|5.2|5.2|5.2|5.2|5.2|5.3|5.3|5.3".split("|")
  },
  2011: {
    serie: "5|5.3|5.1|5.2|5.4|5.6|5.2|5.3|5.1|4.9|5.1|5".split("|"),
    tendencia: "5.2|5.2|5.2|5.3|5.3|5.3|5.3|5.2|5.1|5|5|5".split("|")
  },
  2012: {
    serie: "4.6|5.3|5|4.9|4.8|4.8|4.8|4.9|4.6|4.9|5.2|4.9".split("|"),
    tendencia: "5|5|5|4.9|4.9|4.8|4.8|4.8|4.9|4.9|5|5".split("|")
  },
  2013: {
    serie: "5.2|4.8|5|5|5|5|4.9|4.8|5|4.9|4.6|4.7".split("|"),
    tendencia: "5|5|5|5|5|5|4.9|4.9|4.8|4.8|4.8|4.8".split("|")
  },
  2014: {
    serie: "4.9|4.8|5.3|4.9|5|4.8|5.2|4.8|4.8|4.7|4.7|4.1".split("|"),
    tendencia: "4.8|4.8|4.9|4.9|4.9|4.9|4.9|4.8|4.8|4.7|4.6|4.5".split("|")
  },
  2015: {
    serie: "4.4|4.4|4.3|4.3|4.4|4.4|4.5|4.4|4.2|4.4|4.1|4.3".split("|"),
    tendencia: "4.4|4.4|4.4|4.4|4.4|4.4|4.4|4.4|4.3|4.3|4.3|4.3".split("|")
  },
  2016: {
    serie: "4.1|4.3|4.1|3.8|4|3.9|3.8|3.8|3.9|3.6|3.6|3.6".split("|"),
    tendencia: "4.2|4.2|4.1|4|4|3.9|3.9|3.8|3.7|3.7|3.6|3.6".split("|")
  },
  2017: {
    serie: "3.5|3.5|3.5|3.5|3.6|3.3|3.3|3.3|3.4|3.4|3.5|3.3".split("|"),
    tendencia: "3.5|3.5|3.5|3.4|3.4|3.3|3.3|3.3|3.4|3.4|3.4|3.4".split("|")
  },
  2018: {
    serie: "3.3|3.3|3.2|3.4|3.2|3.4|3.3|3.3|3.4|3.2|3.3|3.5".split("|"),
    tendencia: "3.3|3.3|3.3|3.3|3.3|3.3|3.3|3.3|3.3|3.4|3.4|3.4".split("|")
  },
  2019: {
    serie: "3.5|3.4|3.6|3.5|3.5|3.6|3.6|3.6|3.5|4.7|4.7|4.1".split("|"),
    tendencia: "3.4|3.5|3.5|3.5|3.5|3.5|3.6|3.6|3.6|5|5|5".split("|")
  }
};

export default data;

export const meses = [
  "Ene",
  "Feb",
  "Mar",
  "Abr",
  "May",
  "Jun",
  "Jul",
  "Ago",
  "Sep",
  "Oct",
  "Nov",
  "Dic"
];
