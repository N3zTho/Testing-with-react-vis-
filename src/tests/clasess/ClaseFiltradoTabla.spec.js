import React, { useState } from "react";
import { create, act } from "react-test-renderer";
import data from "../../helpers/data";
import reducers from "../../redux/reducer";
import Tablas from "../../components/tabla/Tablas";

const BotonMostrar = ({ years }) => {
  return <Tablas years={years} />;
};

it("as a u user I can show filtered years", () => {
  const state = {
    years: data,
    filtered: []
  };

  const payload = {
    start: {
      year: 2016
    },
    end: {
      year: 2019
    }
  };

  const reducer = reducers(state, { type: "GET_YEARS_FILTERED", payload });

  let component;

  act(() => {
    component = create(<BotonMostrar years={reducer.filtered} />);
  });

  expect(component.toJSON()).toMatchSnapshot();
});
