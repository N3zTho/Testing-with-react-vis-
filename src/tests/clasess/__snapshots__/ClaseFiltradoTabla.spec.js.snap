// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`as a u user I can show filtered years 1`] = `
<div
  className="table-responsive"
>
  <table
    className="table table-borderless table-sm"
  >
    <tbody>
      <tr>
        <td>
          <table
            className="table table-bordered"
          >
            <thead
              className="thead-light"
            >
              <tr>
                <th>
                   
                </th>
              </tr>
              <tr>
                <th>
                  Denominacion
                </th>
              </tr>
              <tr>
                <td>
                  Serie desestacionalizada
                </td>
              </tr>
              <tr>
                <td>
                  Tendencia ciclo
                </td>
              </tr>
            </thead>
          </table>
        </td>
        <td>
          <table
            className="table table-bordered"
          >
            <thead
              className="thead-light"
            >
              <tr>
                <th
                  className="text-center"
                  colSpan={12}
                >
                  2016
                </th>
              </tr>
              <tr>
                <th
                  scope="col"
                >
                  Ene
                </th>
                <th
                  scope="col"
                >
                  Feb
                </th>
                <th
                  scope="col"
                >
                  Mar
                </th>
                <th
                  scope="col"
                >
                  Abr
                </th>
                <th
                  scope="col"
                >
                  May
                </th>
                <th
                  scope="col"
                >
                  Jun
                </th>
                <th
                  scope="col"
                >
                  Jul
                </th>
                <th
                  scope="col"
                >
                  Ago
                </th>
                <th
                  scope="col"
                >
                  Sep
                </th>
                <th
                  scope="col"
                >
                  Oct
                </th>
                <th
                  scope="col"
                >
                  Nov
                </th>
                <th
                  scope="col"
                >
                  Dic
                </th>
              </tr>
            </thead>
            <tbody>
              <tr>
                <td>
                  4.1
                </td>
                <td>
                  4.3
                </td>
                <td>
                  4.1
                </td>
                <td>
                  3.8
                </td>
                <td>
                  4
                </td>
                <td>
                  3.9
                </td>
                <td>
                  3.8
                </td>
                <td>
                  3.8
                </td>
                <td>
                  3.9
                </td>
                <td>
                  3.6
                </td>
                <td>
                  3.6
                </td>
                <td>
                  3.6
                </td>
              </tr>
              <tr>
                <td>
                  4.2
                </td>
                <td>
                  4.2
                </td>
                <td>
                  4.1
                </td>
                <td>
                  4
                </td>
                <td>
                  4
                </td>
                <td>
                  3.9
                </td>
                <td>
                  3.9
                </td>
                <td>
                  3.8
                </td>
                <td>
                  3.7
                </td>
                <td>
                  3.7
                </td>
                <td>
                  3.6
                </td>
                <td>
                  3.6
                </td>
              </tr>
            </tbody>
          </table>
        </td>
      </tr>
    </tbody>
  </table>
  <table
    className="table table-borderless table-sm"
  >
    <tbody>
      <tr>
        <td>
          <table
            className="table table-bordered"
          >
            <thead
              className="thead-light"
            >
              <tr>
                <th>
                   
                </th>
              </tr>
              <tr>
                <th>
                  Denominacion
                </th>
              </tr>
              <tr>
                <td>
                  Serie desestacionalizada
                </td>
              </tr>
              <tr>
                <td>
                  Tendencia ciclo
                </td>
              </tr>
            </thead>
          </table>
        </td>
        <td>
          <table
            className="table table-bordered"
          >
            <thead
              className="thead-light"
            >
              <tr>
                <th
                  className="text-center"
                  colSpan={12}
                >
                  2017
                </th>
              </tr>
              <tr>
                <th
                  scope="col"
                >
                  Ene
                </th>
                <th
                  scope="col"
                >
                  Feb
                </th>
                <th
                  scope="col"
                >
                  Mar
                </th>
                <th
                  scope="col"
                >
                  Abr
                </th>
                <th
                  scope="col"
                >
                  May
                </th>
                <th
                  scope="col"
                >
                  Jun
                </th>
                <th
                  scope="col"
                >
                  Jul
                </th>
                <th
                  scope="col"
                >
                  Ago
                </th>
                <th
                  scope="col"
                >
                  Sep
                </th>
                <th
                  scope="col"
                >
                  Oct
                </th>
                <th
                  scope="col"
                >
                  Nov
                </th>
                <th
                  scope="col"
                >
                  Dic
                </th>
              </tr>
            </thead>
            <tbody>
              <tr>
                <td>
                  3.5
                </td>
                <td>
                  3.5
                </td>
                <td>
                  3.5
                </td>
                <td>
                  3.5
                </td>
                <td>
                  3.6
                </td>
                <td>
                  3.3
                </td>
                <td>
                  3.3
                </td>
                <td>
                  3.3
                </td>
                <td>
                  3.4
                </td>
                <td>
                  3.4
                </td>
                <td>
                  3.5
                </td>
                <td>
                  3.3
                </td>
              </tr>
              <tr>
                <td>
                  3.5
                </td>
                <td>
                  3.5
                </td>
                <td>
                  3.5
                </td>
                <td>
                  3.4
                </td>
                <td>
                  3.4
                </td>
                <td>
                  3.3
                </td>
                <td>
                  3.3
                </td>
                <td>
                  3.3
                </td>
                <td>
                  3.4
                </td>
                <td>
                  3.4
                </td>
                <td>
                  3.4
                </td>
                <td>
                  3.4
                </td>
              </tr>
            </tbody>
          </table>
        </td>
      </tr>
    </tbody>
  </table>
  <table
    className="table table-borderless table-sm"
  >
    <tbody>
      <tr>
        <td>
          <table
            className="table table-bordered"
          >
            <thead
              className="thead-light"
            >
              <tr>
                <th>
                   
                </th>
              </tr>
              <tr>
                <th>
                  Denominacion
                </th>
              </tr>
              <tr>
                <td>
                  Serie desestacionalizada
                </td>
              </tr>
              <tr>
                <td>
                  Tendencia ciclo
                </td>
              </tr>
            </thead>
          </table>
        </td>
        <td>
          <table
            className="table table-bordered"
          >
            <thead
              className="thead-light"
            >
              <tr>
                <th
                  className="text-center"
                  colSpan={12}
                >
                  2018
                </th>
              </tr>
              <tr>
                <th
                  scope="col"
                >
                  Ene
                </th>
                <th
                  scope="col"
                >
                  Feb
                </th>
                <th
                  scope="col"
                >
                  Mar
                </th>
                <th
                  scope="col"
                >
                  Abr
                </th>
                <th
                  scope="col"
                >
                  May
                </th>
                <th
                  scope="col"
                >
                  Jun
                </th>
                <th
                  scope="col"
                >
                  Jul
                </th>
                <th
                  scope="col"
                >
                  Ago
                </th>
                <th
                  scope="col"
                >
                  Sep
                </th>
                <th
                  scope="col"
                >
                  Oct
                </th>
                <th
                  scope="col"
                >
                  Nov
                </th>
                <th
                  scope="col"
                >
                  Dic
                </th>
              </tr>
            </thead>
            <tbody>
              <tr>
                <td>
                  3.3
                </td>
                <td>
                  3.3
                </td>
                <td>
                  3.2
                </td>
                <td>
                  3.4
                </td>
                <td>
                  3.2
                </td>
                <td>
                  3.4
                </td>
                <td>
                  3.3
                </td>
                <td>
                  3.3
                </td>
                <td>
                  3.4
                </td>
                <td>
                  3.2
                </td>
                <td>
                  3.3
                </td>
                <td>
                  3.5
                </td>
              </tr>
              <tr>
                <td>
                  3.3
                </td>
                <td>
                  3.3
                </td>
                <td>
                  3.3
                </td>
                <td>
                  3.3
                </td>
                <td>
                  3.3
                </td>
                <td>
                  3.3
                </td>
                <td>
                  3.3
                </td>
                <td>
                  3.3
                </td>
                <td>
                  3.3
                </td>
                <td>
                  3.4
                </td>
                <td>
                  3.4
                </td>
                <td>
                  3.4
                </td>
              </tr>
            </tbody>
          </table>
        </td>
      </tr>
    </tbody>
  </table>
  <table
    className="table table-borderless table-sm"
  >
    <tbody>
      <tr>
        <td>
          <table
            className="table table-bordered"
          >
            <thead
              className="thead-light"
            >
              <tr>
                <th>
                   
                </th>
              </tr>
              <tr>
                <th>
                  Denominacion
                </th>
              </tr>
              <tr>
                <td>
                  Serie desestacionalizada
                </td>
              </tr>
              <tr>
                <td>
                  Tendencia ciclo
                </td>
              </tr>
            </thead>
          </table>
        </td>
        <td>
          <table
            className="table table-bordered"
          >
            <thead
              className="thead-light"
            >
              <tr>
                <th
                  className="text-center"
                  colSpan={12}
                >
                  2019
                </th>
              </tr>
              <tr>
                <th
                  scope="col"
                >
                  Ene
                </th>
                <th
                  scope="col"
                >
                  Feb
                </th>
                <th
                  scope="col"
                >
                  Mar
                </th>
                <th
                  scope="col"
                >
                  Abr
                </th>
                <th
                  scope="col"
                >
                  May
                </th>
                <th
                  scope="col"
                >
                  Jun
                </th>
                <th
                  scope="col"
                >
                  Jul
                </th>
                <th
                  scope="col"
                >
                  Ago
                </th>
                <th
                  scope="col"
                >
                  Sep
                </th>
                <th
                  scope="col"
                >
                  Oct
                </th>
                <th
                  scope="col"
                >
                  Nov
                </th>
                <th
                  scope="col"
                >
                  Dic
                </th>
              </tr>
            </thead>
            <tbody>
              <tr>
                <td>
                  3.5
                </td>
                <td>
                  3.4
                </td>
                <td>
                  3.6
                </td>
                <td>
                  3.5
                </td>
                <td>
                  3.5
                </td>
                <td>
                  3.6
                </td>
                <td>
                  3.6
                </td>
                <td>
                  3.6
                </td>
                <td>
                  3.5
                </td>
              </tr>
              <tr>
                <td>
                  3.4
                </td>
                <td>
                  3.5
                </td>
                <td>
                  3.5
                </td>
                <td>
                  3.5
                </td>
                <td>
                  3.5
                </td>
                <td>
                  3.5
                </td>
                <td>
                  3.6
                </td>
                <td>
                  3.6
                </td>
                <td>
                  3.6
                </td>
              </tr>
            </tbody>
          </table>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
